### ISSUE TITLE
Order by statements cause error in postgres when counting records

### ISSUE THREAD

**Issue by KRTac** [jonathangeiger/kohana-jelly#95]
```
Database_Exception [ 0 ]: ERROR: column "users.username" must appear in the GROUP BY clause or be used in an aggregate function LINE 1: ...users" WHERE "users"."username" = 'user' ORDER BY "users"."u...
[ SELECT COUNT(*) AS "total" FROM "users" WHERE "users"."username" = 'user' ORDER BY "users"."username" ASC ]
```

I discovered this error while trying to add a user to the database. As [banks suggested](http://forum.kohanaphp.com/comments.php?DiscussionID=5303&page=1#Comment_37536), the problem most likely occours while trying to check if a entry is unique. In my case the username entry (and also the email entry, but the username comes first). The problem lies in the ORDER BY part of the query.
On the other hand, the query:

```
SELECT COUNT(*) AS "total" FROM "users" WHERE "users"."username" = 'user' 
```

runs like a charm. It doesn't make sense to order this particular query, so why not just leave the ordering part out entirely?

**Comment by Lokaltog** [comment]
I have the same problem, and I think a possible solution could be to allow `Jelly::select()`   to receive an optional array of columns to fetch for situations like this (I think that also may help solve part of #88).

This could have other benefits as well, e.g. if you don't need all the columns from a particular query, it will save resources to only load the columns you need.

An example:

```
Jelly::select('model', null, array('only', 'fetch', 'these', 'columns'))
// Or something like this:
Jelly::select('model')->limit_columns(array('only', 'fetch', 'these', 'columns'))
```

This kind of limitation could be used in `Jelly_Field_Core::_is_unique()` to avoid this issue with postgres.

**Comment by jonathangeiger** [comment]
I'm going to fix this rather soon but I want to get Postgres into the test suite first, so it's taking a bit longer.

**Comment by KRTac** [comment]
What DB driver for postgres will you use? [This one](http://github.com/cbandy/kohana-postgresql) maybe?

**Comment by jonathangeiger** [comment]
Ideally I can just use the PDO driver that's shipped with the database module. Jelly doesn't depend on any vendor specific extensions.

**Comment by jonathangeiger** [comment]
Closed by fccdf1eee766654a5b9cb86d5936491dff7a2e9a

**Comment by jonathangeiger** [comment]
Thanks for your help everyone. I ended up having to use the driver KRTac recommended because there are a few outstanding bugs with using just the PDO driver.

I also came across [another bug](http://github.com/jonathangeiger/kohana-jelly/issues#issue/109) as a result of setting up these tests. If anyone can offer any insight I'd be very happy! :)
